packages/ckeditor5-indent/src/indentblockcommand.ts

```diff
@@ -92,10 +94,17 @@ export default class IndentBlockCommand extends Command {
 /**
  * Returns blocks from selection that should have blockIndent selection set.
  */
-function getBlocksToChange( model: Model ): Array<Element> {
+function getBlocksToChange( model: Model, isForward: boolean ): Array<Element> {
 	const selection = model.document.selection;
 	const schema = model.schema;
 	const blocksInSelection = Array.from( selection.getSelectedBlocks() );
 
-	return blocksInSelection.filter( block => schema.checkAttribute( block, 'blockIndent' ) );
+	return blocksInSelection.filter( block => {
+		// Do not add blockIndent to block items in Document List items. See https://github.com/ckeditor/ckeditor5/issues/14155.
+		if ( isForward ) {
+			return schema.checkAttribute( block, 'blockIndent' ) && !block.hasAttribute( 'listItemId' );
```

Comment:
Maybe we could use a helper to check if it's a list item block (and share it with `refresh` check).

Maybe sth like `canBeIndented()`?

Reply:
Done.